statistics/handle/lru_cache.go

```diff
@@ -0,0 +1,334 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package handle
+
+import (
+	"container/list"
+	"sync"
+
+	"github.com/pingcap/tidb/metrics"
+	"github.com/pingcap/tidb/statistics"
+)
+
+const (
+	typHit    = "hit"
+	typMiss   = "miss"
+	typUpdate = "update"
+	typDel    = "del"
+	typEvict  = "evict"
+	typCopy   = "copy"
+	typTrack  = "track"
+	typTotal  = "total"
+)
+
+// cacheEntry wraps Key and Value. It's the value of list.Element.
+type cacheItem struct {
+	key         int64
+	value       *statistics.Table
+	tblMemUsage *statistics.TableMemoryUsage
+}
+
+// internalLRUCache is a simple least recently used cache
+type internalLRUCache struct {
```

Comment:
We have some cache implemented in the tidb repo like ```util/kvcache/simple_lru.go``` and ```github.com/dgraph-io/ristretto```.

Do we have to implement an LRU again?

Reply:
Yes, because the evict policy is dedicated designed for statsCache